pkg/controllers/user/alert/watcher/node.go

```diff
@@ -54,7 +55,7 @@ func (w *NodeWatcher) watchRule() error {
 		return nil
 	}
 
-	clusterAlerts, err := w.clusterAlertLister.List("", labels.NewSelector())
+	clusterAlerts, err := w.clusterAlerts.Controller().Lister().List("", labels.NewSelector())
```

Comment:
I'd use a lister to list object, and introduce a client for only remove. We want to eliminate all calls to API directly unless it's needed